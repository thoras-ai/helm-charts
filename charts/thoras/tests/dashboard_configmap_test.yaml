suite: Dashboard
templates:
  - dashboard/nginx-config-map.yaml
tests:
  - it: Should default to v2 port
    set:
      thorasVersion: TEST
    asserts:
      - matchSnapshot:
        path: data.nginx.conf
  - it: Should use the v1 port if v2 is disabled
    set:
      thorasVersion: TEST
      thorasDashboard:
        v2:
          enabled: false
    asserts:
      - matchSnapshot:
        path: data.nginx.conf
