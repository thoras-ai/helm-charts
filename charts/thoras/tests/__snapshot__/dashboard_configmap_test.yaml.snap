Should default to v2 port:
  1: |
    apiVersion: v1
    data:
      dashboard-v2.nginx.conf: |
        events {
          worker_connections 1024;
        }

        http {
          include       mime.types;

          server {
            listen       5173;
            server_name  localhost;

            root   /usr/share/nginx/html;
            index  index.html index.htm;

            location / {
              try_files $uri /index.html;
            }

            location ~* \.(js|css|png|jpg|jpeg|gif|ico|svg|woff|woff2|ttf|eot|otf|ttf|map)$ {
              expires 1y;
              access_log off;
              add_header Cache-Control "public";
            }

            error_page   500 502 503 504  /50x.html;
            location = /50x.html {
              root   /usr/share/nginx/html;
            }
          }
        }
      nginx.conf: |
        events {
          worker_connections 1024;
        }

        http {
          server {
            listen 80;

            location /v1/ {
              proxy_pass http://thoras-api-server-v2;
              proxy_set_header Host $host;
              proxy_set_header X-Real-IP $remote_addr;
              proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
              proxy_set_header X-Forwarded-Proto $scheme;
            }

            location /config.json {
                default_type application/json;
                return 200 '{ "api_base_url": "", "extras": {"cluster_name":"phantom-assassin"} }';
            }

            location / {
              proxy_pass http://localhost:5173;
              proxy_set_header Host $host;
              proxy_set_header X-Real-IP $remote_addr;
              proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
              proxy_set_header X-Forwarded-Proto $scheme;       }
          }
        }
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: thoras
        helm.sh/chart: thoras-4.36.0
      name: thoras-dashboard-nginx-config
      namespace: NAMESPACE
Should use the v1 port if v2 is disabled:
  1: |
    apiVersion: v1
    data:
      dashboard-v2.nginx.conf: |
        events {
          worker_connections 1024;
        }

        http {
          include       mime.types;

          server {
            listen       5173;
            server_name  localhost;

            root   /usr/share/nginx/html;
            index  index.html index.htm;

            location / {
              try_files $uri /index.html;
            }

            location ~* \.(js|css|png|jpg|jpeg|gif|ico|svg|woff|woff2|ttf|eot|otf|ttf|map)$ {
              expires 1y;
              access_log off;
              add_header Cache-Control "public";
            }

            error_page   500 502 503 504  /50x.html;
            location = /50x.html {
              root   /usr/share/nginx/html;
            }
          }
        }
      nginx.conf: |
        events {
          worker_connections 1024;
        }

        http {
          server {
            listen 80;

            location /v1/ {
              proxy_pass http://thoras-api-server-v2;
              proxy_set_header Host $host;
              proxy_set_header X-Real-IP $remote_addr;
              proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
              proxy_set_header X-Forwarded-Proto $scheme;
            }

            location /config.json {
                default_type application/json;
                return 200 '{ "api_base_url": "", "extras": {"cluster_name":"phantom-assassin"} }';
            }

            location / {
              proxy_pass http://localhost:3000;
              proxy_set_header Host $host;
              proxy_set_header X-Real-IP $remote_addr;
              proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
              proxy_set_header X-Forwarded-Proto $scheme;       }
          }
        }
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: thoras
        helm.sh/chart: thoras-4.36.0
      name: thoras-dashboard-nginx-config
      namespace: NAMESPACE
should set extras in config.json:
  1: |
    apiVersion: v1
    data:
      dashboard-v2.nginx.conf: |
        events {
          worker_connections 1024;
        }

        http {
          include       mime.types;

          server {
            listen       5173;
            server_name  localhost;

            root   /usr/share/nginx/html;
            index  index.html index.htm;

            location / {
              try_files $uri /index.html;
            }

            location ~* \.(js|css|png|jpg|jpeg|gif|ico|svg|woff|woff2|ttf|eot|otf|ttf|map)$ {
              expires 1y;
              access_log off;
              add_header Cache-Control "public";
            }

            error_page   500 502 503 504  /50x.html;
            location = /50x.html {
              root   /usr/share/nginx/html;
            }
          }
        }
      nginx.conf: |
        events {
          worker_connections 1024;
        }

        http {
          server {
            listen 80;

            location /v1/ {
              proxy_pass http://thoras-api-server-v2;
              proxy_set_header Host $host;
              proxy_set_header X-Real-IP $remote_addr;
              proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
              proxy_set_header X-Forwarded-Proto $scheme;
            }

            location /config.json {
                default_type application/json;
                return 200 '{ "api_base_url": "", "extras": {"cluster_name":"phantom-assassin","feature":true,"foo":"bar"} }';
            }

            location / {
              proxy_pass http://localhost:5173;
              proxy_set_header Host $host;
              proxy_set_header X-Real-IP $remote_addr;
              proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
              proxy_set_header X-Forwarded-Proto $scheme;       }
          }
        }
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: thoras
        helm.sh/chart: thoras-4.36.0
      name: thoras-dashboard-nginx-config
      namespace: NAMESPACE
